### PR TITLE
Exercise04

### PR DESCRIPTION
**Pull request by huynhngocminh2511**

### CODE REVIEW

File: Exercise04/Adapters/MyAdapter.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        public MyAdapter(List<Item> items)
+        {
+            this.items = items;
+        }
+
+        // Create new views (invoked by the layout manager)
+        public override RecyclerView.ViewHolder OnCreateViewHolder(ViewGroup parent, int viewType)
+        {
+
+            //Setup your layout here
```
Comment (voquanghoa) [inline Exercise04/Adapters/MyAdapter.cs]: Remove dead comment

File: Exercise04/Adapters/MyAdapter.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            return new MyAdapterViewHolder(itemView);
+        }
+
+        // Replace the contents of a view (invoked by the layout manager)
+        public override void OnBindViewHolder(RecyclerView.ViewHolder viewHolder, int position)
+        {
+            var item = items[position];
+
+            // Replace the contents of the view with that element
+            var holder = (MyAdapterViewHolder)viewHolder;
```
Comment (voquanghoa) [inline Exercise04/Adapters/MyAdapter.cs]: Move this logic to `MyAdapterViewHolder`

File: Exercise04/Models/Item.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+using Android.App;
+using Android.Content;
+using Android.OS;
+using Android.Runtime;
+using Android.Views;
+using Android.Widget;
+
+namespace Exercise04.Models
+{
+    class Item
```
Comment (voquanghoa) [inline Exercise04/Models/Item.cs]: Bad name, make it public

File: Exercise04/Adapters/MyAdapter.cs (hunk 1 above)
Comment (huynhngocminh2511) [reply Exercise04/Adapters/MyAdapter.cs]: Done

File: Exercise04/Adapters/MyAdapter.cs (hunk 2 above)
Comment (huynhngocminh2511) [reply Exercise04/Adapters/MyAdapter.cs]: Done

File: Exercise04/Models/Item.cs (hunk 3 above)
Comment (huynhngocminh2511) [reply Exercise04/Models/Item.cs]: Done

File: Exercise04/Adapters/AdapterAppBookGame.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+using Android.Support.V7.Widget;
+using System.Collections.Generic;
+using Exercise044.Models;
+using Android.App;
+
+namespace Exercise044.Adapters
+{
+    class AdapterAppBookGame : RecyclerView.Adapter
+    {
+        private List<AppBookGame> AppBookGames;
```
Comment (voquanghoa) [inline Exercise04/Adapters/AdapterAppBookGame.cs]: Wrong convention

File: Exercise04/Adapters/AdapterAppBookGame.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        {
+            ((AdapterViewHolderAppBookGames)viewHolder).AppBookGame = AppBookGames[position];
+        }
+
+        public override int ItemCount => AppBookGames.Count;
+    }
+
+    public class AdapterViewHolderAppBookGames : RecyclerView.ViewHolder
+    {
+        private TextView TextViewName;
```
Comment (voquanghoa) [inline Exercise04/Adapters/AdapterAppBookGame.cs]: Wrong convention

File: Exercise04/MainActivity.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                var fragmentManager = FragmentManager.BeginTransaction();
+                fragmentManager.Replace(Resource.Id.frameLayout1, new FragmentBook());
+                fragmentManager.Commit();
+            };
+
+            FindViewById<Button>(Resource.Id.btn_game).Click += delegate
+            {
+                var fragmentManager = FragmentManager.BeginTransaction();
+                fragmentManager.Replace(Resource.Id.frameLayout1, new FragmentGame());
+                fragmentManager.Commit();
```
Comment (voquanghoa) [inline Exercise04/MainActivity.cs]: Make a method
```C#
private void DisplayFragment<T>(){
    var fragmentManager = FragmentManager.BeginTransaction();
                fragmentManager.Replace(Resource.Id.frameLayout1, Activator.CreateInstance<T>());
               fragmentManager.Commit();
}
```
and use
```C#
DisplayFragment< FragmentGame>();
```

Comment (huynhngocminh2511) [reply Exercise04/MainActivity.cs]: Done

File: Exercise04/Adapters/AdapterAppBookGame.cs (hunk 5 above)
Comment (huynhngocminh2511) [reply Exercise04/Adapters/AdapterAppBookGame.cs]: Done

File: Exercise04/Adapters/AdapterAppBookGame.cs (hunk 4 above)
Comment (huynhngocminh2511) [reply Exercise04/Adapters/AdapterAppBookGame.cs]: Done

**Pull request merged**